src/main/java/org/broadinstitute/hellbender/utils/io/IOUtils.java

```diff
@@ -385,4 +394,26 @@ public static PrintStream makePrintStreamMaybeGzipped(File file) throws IOExcept
             return new PrintStream(file);
         }
     }
+
+    /**
+     * Creates a temp file that will be deleted on exit after tests are complete.
```

Comment:
And document the fact that companion Tribble/Tabix/BAM indices to the temp file will also be marked for deletion on exit.